python312Packages.setuptools: 70.0.0 -> 72.1.0

mweinelt opened the pull request:
https://setuptools.pypa.io/en/stable/history.html#v72-1-0

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
python312Packages.setuptoolsCheckHook: remove

The hook relied on the `test` command passed to `setup.py`, which has
long been deprecated and finally removed in setuptools 72.0.

mweinelt committed:
python312Packages.glyphtools: disable tests

The setup.py test command has been removed. This package had no tests in
the first place, so we reach partiy by import testing the module.

mweinelt committed:
whipper: test with pytest

mweinelt committed:
python312Packages.pycdio: fix tests on 3.12

mweinelt committed:
python312Packages.setuptools: 70.0.0 -> 72.1.0

https://setuptools.pypa.io/en/stable/history.html#v72-1-0

mweinelt committed:
python312Packages.apsw: test with pytest

mweinelt committed:
python312Packages.serpent: test with pytest

SuperSandro2000 commented on pkgs/development/python-modules/glyphtools/default.nix:
```diff
@@ -26,11 +25,9 @@ buildPythonPackage rec {
 
   # A unit test suite does exist, but it only contains a dummy test that
   # doesn't test anything. It does import the module though so we still run it.
```
```suggestion
  # A unit test suite does exist, but it only contains a dummy test that
  # doesn't test anything.
```

SuperSandro2000 commented on pkgs/development/python-modules/pycdio/default.nix:
```diff
@@ -23,6 +23,8 @@ buildPythonPackage rec {
       --replace 'library_dirs=library_dirs' 'library_dirs=[dir.decode("utf-8") for dir in library_dirs]' \
       --replace 'include_dirs=include_dirs' 'include_dirs=[dir.decode("utf-8") for dir in include_dirs]' \
       --replace 'runtime_library_dirs=runtime_lib_dirs' 'runtime_library_dirs=[dir.decode("utf-8") for dir in runtime_lib_dirs]'
+    substituteInPlace test/test-cdtext.py \
+      --replace assertEquals assertEqual
```
```suggestion
      --replace-fail 'library_dirs=library_dirs' 'library_dirs=[dir.decode("utf-8") for dir in library_dirs]' \
      --replace-fail 'include_dirs=include_dirs' 'include_dirs=[dir.decode("utf-8") for dir in include_dirs]' \
      --replace-fail 'runtime_library_dirs=runtime_lib_dirs' 'runtime_library_dirs=[dir.decode("utf-8") for dir in runtime_lib_dirs]'
    substituteInPlace test/test-cdtext.py \
      --replace-fail assertEquals assertEqual
```

mweinelt committed:
python312Packages.intervaltree: migrate to pytest and pep517 lingo

mweinelt committed:
python312Packages.scikit-build: 0.17.6 -> 0.18.0

https://github.com/scikit-build/scikit-build/releases/tag/0.18.0

mweinelt committed:
python312Packages.pytest-fixture-config: patch out tests_require

The source is also reused in pytest-server-fixtures, pytest-shutil and
pytest-virtualenv, which now inherit the patch.

mweinelt committed:
python312Packages.rapidfuzz: 3.9.4 -> 3.9.5

https://github.com/maxbachmann/RapidFuzz/blob/refs/tags/v3.9.5/CHANGELOG.rst

mweinelt committed:
python312Packages.pystray: run tests with pytest

mweinelt committed:
python312Packages.myst-parser: disable failing tests

mweinelt committed:
python312Packages.pdm-backend: 2.3.1 -> 2.3.3

https://github.com/pdm-project/pdm-backend/releases/tag/2.3.2
https://github.com/pdm-project/pdm-backend/releases/tag/2.3.3

natsukium commented:
Does this PR turn into python-updates? I will also make time to work on it if necessary.

mweinelt commented:
Eventually, but I only have time for a small run.

mweinelt committed:
electrum-ltc: disable failing and stuck tests

These tests have forced the hydra job for electrum-ltc to run into a
timeout.

mweinelt committed:
python312Packages.datasette: ignore deprecation warnings

mweinelt committed:
python312Packages.python-daemon: fix build

Accessing packaging via setuptools.extern is dead.

mweinelt committed:
python312Packages.liblarch: test with pytest

mweinelt commented:
Now all that is left to do is wait for trofi to find out what else this change broke. :smile:
